courseware/tasks.py

```diff
@@ -1,12 +1,20 @@
 """Courseware tasks"""
-from django.contrib.auth import get_user_model
 
 from mitxpro.celery import app
-from courseware.api import create_edx_user
+from courseware.api import create_edx_user, create_edx_auth_token
+from users.api import get_user_by_id
 
 
 @app.task()
 def create_edx_user_from_id(user_id):
     """Loads user by id and calls the API method to create the user in edX"""
-    user = get_user_model().objects.get(id=user_id)
+    user = get_user_by_id(user_id)
     create_edx_user(user)
+    create_edx_api_auth_for_user.delay(user_id)
```

Comment:
Does it make more sense just to call `create_edx_user` and `create_edx_api_auth_for_user` in series in this task rather than kicking off a task from within this task?

Reply:
I intentionally kept them separate because if the task retries on the latter operation, `create_edx_user` is not idempotent, although i suppose I could make it so it is so we can do this all in one task.